fe/fe-core/src/main/java/org/apache/doris/common/NereidsSqlCacheManager.java

```diff
@@ -396,7 +396,8 @@ private boolean usedVariablesChanged(List<Variable> currentVariables, SqlCacheCo
             Variable currentVariable = currentVariables.get(i);
             Variable cachedVariable = cachedUsedVariables.get(i);
             if (!Objects.equals(currentVariable, cachedVariable)
-                    || cachedVariable.getRealExpression().anyMatch(Nondeterministic.class::isInstance)) {
+                    || cachedVariable.getRealExpression().anyMatch(expr -> expr instanceof ExpressionTrait
+                    && !((ExpressionTrait) expr).isDeterministic())) {
```

Comment:
You can remove `expr instanceof ExpressionTrait` because Expression is always ExpressionTrait